src/ivoc/symdir.cpp

```diff
@@ -60,32 +61,16 @@ class SymDirectoryImpl: public Observer {
     void sort();
 };
 
-static int compare_entries(const void* k1, const void* k2) {
-    SymbolItem* e1 = *((SymbolItem**) k1);
-    SymbolItem* e2 = *((SymbolItem**) k2);
+static int compare_entries(const SymbolItem* e1, const SymbolItem* e2) {
     int i = strcmp(e1->name().string(), e2->name().string());
     if (i == 0) {
-        if (e1->array_index() > e2->array_index()) {
-            i = 1;
-        } else {
-            i = -1;
-        }
+        return e1->array_index() > e2->array_index();
```

Comment:
Is this correct? Say `i == 0` and `e1->array_index() > e2->array_index() == false` then the previous implementation would return `-1` while the proposed implementation would return `-1`.

Reply:
the previous implementation was based on `qsort()`. The comparator should return `-1`, `0' or '1'.
The new one `std::sort` use `true` or `false`.